src/app/3d/qgs3danimationwidget.cpp

```diff
@@ -175,6 +187,67 @@ void Qgs3DAnimationWidget::onAnimationTimer()
   }
 }
 
+void Qgs3DAnimationWidget::onExportAnimation()
+{
+  if ( !mMap || !mAnimationSettings )
+    QMessageBox::warning( this, tr( "Export Animation" ), tr( "Unable to export 3D animation" ) );
+
+  Qgs3DAnimationExportDialog dialog;
+  if ( dialog.exec() == QDialog::Accepted )
+  {
+    Qgs3DAnimationSettings animationSettings = animation();
```

Comment:
Could this code be moved outside the widget and app and into a reusable method in the 3d API?